Java/core/src/main/java/com/amazon/randomcutforest/tree/Node.java

```diff
@@ -422,4 +422,24 @@ protected void deleteSequenceIndex(long sequenceIndex) {
             sequenceIndexes.remove(sequenceIndex);
         }
     }
+
+    /**
+     * The following function returns the bounding box corresponding to the child
+     * that is not the path from the node to leaf traversed by a point. If such a
+     * box cannot be defined then the function returns null.
+     * 
+     * @param point the point as above
+     * @param node  the node as above
+     * @return the bounding box of the node
+     */
+
+    public static BoundingBox getSiblingBoundingBox(double[] point, Node node) {
+        if (node == null)
```

Comment:
Should this method every be called when node is null? Maybe this should throw an exception (e.g. using `checkNotNull` instead of silently returning null.

Reply:
True. The function should not be called on a null node. Will fix.